### ISSUE TITLE
Consider renaming ReceivingAddress to ReceivingKey.

### ISSUE THREAD

**Issue by dan-da** [Neptune-Crypto/neptune-core#277]
See the discussion in #270.   In particular, https://github.com/Neptune-Crypto/neptune-core/pull/270#issuecomment-2516400503 and preceding comments for rationale.   Filing this issue now so its not just forgotten.

@aszepieniec may have other naming suggestion(s).

If done, it should be a global replace including:
 * struct + enum + variable + function + rpc names. 
 * strings: UI labels, arguments, error messages
 * docs: readme, manual, etc.

Does not seem a very high priority except that if we are going to do it, then it should be done before mainnet to avoid confusing users.

**Comment by aszepieniec** [comment]
Recent commits merged into master in d2f6ecf77a06c73b487b4457f203cefbd16cc30a add more context here.

There are UTXOs that come into existence without any prior transfers of funds: guesser fee UTXOs. The guesser should be able to claim custody of these guesser fee UTXOs without ever generating an address for them.

Guessing works as follows. The guesser samples a random nonce-preimage, and hashes it. The hash is the nonce, which goes into the block header. If the hash of the block is smaller than the target, the guesser wins. The resulting block generates two UTXOs, one time-locked and one liquid immediately. Both are locked by a lock script that can be satisfied only by the nonce-preimage.

In order to be able to spend guesser fee UTXOs, the wallet must be made aware of their unlock keys. Prior to this branch being merged, there was no data structure or logical pathway to account for this. The wallet *was* monitoring the guesser UTXOs, because the guesser UTXOs *were* being added to the expected utxos list, recognized, and ultimately added to the `MonitoredUtxo`s database with the right sender randomness and receiver preimage. So the wallet was showing the right balance but when you tried to spend it you would get an error complaining about no matching key being found.

We solved this issue by adding a variant `RawHashLock` to `KeyType` and `SpendingKey`. The new variant breaks the correspondence between `SpendingKey`s and `ReceivingAddress`es, because now there is a variant of `SpendingKey` that has no matching `ReceivingAddress`.

**Comment by Sword-Smith** [comment]
With the introduction of the `SpendingKey` variant `RawHashLock`  in adee45c8211b5d67b3a6625309a27ba61ff7e3cb, it seems like there is a spending key that doesn't have an associated receiving address. The following vocabulary should show why I think receiving keys and addresses are two different things:

In the context of a lock script whose successful execution proves the knowledge of a hash-preimage, we can define:
- The spending key: The preimage required to successfully execute the program
- The receiving key: Either the lock script-hash or the after-image
- The address: The lock script, receiver digest, cryptographic (public) keys to encrypt information to the receiver, plus potential receiver identifiers to make it easier for the receiver to identify incoming UTXOs.

The guesser UTXOs only have a receiving key associated with them -- the data that allows all clients to calculate the mutator set after applying the block; the guesser UTXOs do not have an associated address since there is no information required by any sender, as there is no sender.

TL;DR, the address is more than just a public key. It also contains information about how to communicate with the receiver.
